tuner/tuner/common_test.py

```diff
@@ -206,3 +206,142 @@ def test_get_lowering_config(tuner_ctx: common.TunerContext) -> None:
+            -> (!transform.any_op) attributes { iree_codegen.tuning_spec_entrypoint } {
+                %res = transform.foreach_match in %arg0 @match -> @apply_op_config
+                    : (!transform.any_op) -> (!transform.any_op)
+                transform.yield %res : !transform.any_op
+            }
+        }
+    """
+
+    first_ir_module = ir.Module.parse(first_module_str, context)
+    second_ir_module = ir.Module.parse(second_module_str, context)
+    linked_module = common.link_tuning_specs(
+        tuner_ctx, [first_ir_module, second_ir_module]
+    )
+    assert linked_module
+
+    assert "transform.with_named_sequence" in linked_module.operation.attributes
+    assert (
+        "iree_codegen.tuning_spec_with_default_entrypoint"
+        in linked_module.operation.attributes
+    )
+
+    inner_ops = list(linked_module.body.operations)
+    # Check that inner modules have been merged into the top-level module and no inner modules remain.
+    assert all(
+        op.name != "builtin.module" for op in inner_ops
+    ), "Unexpected inner builtin.module ops found"
+
+    named_sequences = []
+    kernel_config_op = None
+    for op in linked_module.body.operations:
+        if op.name == "transform.named_sequence":
+            sym_name_attr = op.sym_name
+            assert sym_name_attr is not None
+            named_sequences.append(sym_name_attr.value)
+            if sym_name_attr.value == "__kernel_config":
+                kernel_config_op = op
+
+    assert kernel_config_op is not None, "Missing @__kernel_config"
+    expected_names = {
+        "match",
+        "apply_op_config",
+        "inner_module_b_match",
+        "inner_module_b_apply_op_config",
+        "__kernel_config",
+    }
+    assert (
+        set(named_sequences) == expected_names
+    ), f"Unexpected named sequence names: {named_sequences}"
+
+    foreach_match_op = kernel_config_op.body.operations[0]
+    assert (
+        foreach_match_op.name == "transform.foreach_match"
+    ), f"Expected op name 'transform.foreach_match', got '{foreach_match_op.name}'"
+    assert (
+        len(foreach_match_op.matchers) == len(foreach_match_op.actions)
+        and len(foreach_match_op.matchers) == 2
+    )
+    assert (
+        foreach_match_op.matchers[0].value == "match"
+    ), f"Expected first matcher to be 'match', got '{foreach_match_op.matchers[0].value}'"
+    assert (
+        foreach_match_op.matchers[1].value == "inner_module_b_match"
+    ), f"Expected second matcher to be 'inner_module_b_match', got '{foreach_match_op.matchers[1].value}'"
+    assert (
+        foreach_match_op.actions[0].value == "apply_op_config"
+    ), f"Expected first action to be 'apply_op_config', got '{foreach_match_op.actions[0].value}'"
+    assert (
+        foreach_match_op.actions[1].value == "inner_module_b_apply_op_config"
+    ), f"Expected second action to be 'inner_module_b_apply_op_config', got '{foreach_match_op.actions[1].value}'"
```

Comment:
We shouldn't need to repeat the same tests for the logic already tested in IREE. Here, I'd only check that that there are no nested modules and that there is a `__kernel_config` op.